File: 2236-maximum-twin-sum-of-a-linked-list/maximum-twin-sum-of-a-linked-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
int pairSum(struct ListNode* head) {
    int count=0;
    struct ListNode* p=head;
    while(p!=NULL){
        count++;
        p=p->next;
    }
    int arr[count];
    p=head;
    for(int i=0;i<count;i++){
        arr[i]=p->val;
        p=p->next;
    }
    int maxSum = 0;  
    for(int i = 0; i < count/2; i++) {  
        int currentSum = arr[i] + arr[count-1-i];
        if(currentSum > maxSum) {
            maxSum = currentSum;
        }
    }
    return maxSum;
}

File: 2236-maximum-twin-sum-of-a-linked-list/maximum-twin-sum-of-a-linked-list.java
class Solution {
    public int pairSum(ListNode head) {
        int count = 0;
        ListNode temp = head;
        while (temp != null) {
            count++;
            temp = temp.next;
        }
        int[] arr = new int[count];
        int i = 0;
        temp = head;
        while (temp != null) {
            arr[i++] = temp.val;
            temp = temp.next;
        }

        int maxSum = 0;
        for (i = 0; i < count / 2; i++) {
            int sum = arr[i] + arr[count - 1 - i];
            maxSum = Math.max(maxSum, sum);
        }

        return maxSum;
    }
}
